File: PharmaCity.Angular/src/models/Purchase.ts
import { Petition } from "./Petition";
import { Pharmacy } from "./Pharmacy";

export class Purchase{

    id: number;
    code: string;
    shopping: Array<Petition>;

    constructor(){
        this.id = 0;
        this.code = "";
        this.shopping = [];
    }
}

File: PharmaCity.System/PharmaCity.Domain/Purchase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PharmaCity.Domain
{
    public class Purchase
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public Pharmacy Pharmacy { get; set; }
        public IEnumerable<PetitionBuy> Shopping { get; set; }
        public State State { get; set; }
    }
}
